src/pages/base.js

```diff
@@ -14,17 +14,27 @@ class Page {
   }
 
   visitWithParameters(data) {
-    let url = this.url;
-
-    for (const item of data.raw()) {
-      url = url.replace(`:${item[0]}`, item[1]);
-    }
-
-    if (config.type === 'otherWeb') {
-      return protractor.browser.get(url);
-    }
-
-    return protractor.browser.get(url).then(() => protractor.browser.waitForAngular());
+    const esc = encodeURIComponent;
+    const additionalParams = [];
+
+    this.url =
+      data.raw().reduce((url, item) => {
+        if (url.indexOf(`:${item[0]}`) === -1) {
+          additionalParams.push(item);
+          return url;
+        }
+        return url.replace(`:${item[0]}`, item[1]);
+      }, this.url) +
+      (additionalParams.length > 0
```

Comment:
All of this is done with `querystring` module https://nodejs.org/api/querystring.html.